.travis.yml

```diff
@@ -1,11 +1,23 @@
 language: rust
 rust:
-  - nightly
-
-before_script: |
-  rustup component add rustfmt-preview
+  - stable
+  - beta
 script: |
-  cargo fmt -- --write-mode=diff &&
   cargo build --verbose &&
   cargo test  --verbose
+matrix: # additional tests
+  include:
+  - rust: nightly-2018-04-13
+    env: CLIPPY_VERS="0.0.193"
+    before_script: |
+      [[ "$(cargo +nightly-2018-04-13 clippy --version)" != "$CLIPPY_VERS" ]] && \
```

Comment:
This actually isn't needed.  I know I recommended it in the past for assert_cli but have learned more about travis as part of documenting it for crate-ci.

Travis will ignore the caches on environment variable changes and since the version is kept in an environment variable, it will auto-clear.

Reply:
Oh, interesting! Should be depend on Travis' behavior here, though? At least this way it should also work for other CI systems, right?

Reply:
Except the file is travis specific.  As we document other CIs, we can note the special things that might need to be done in them.  Overall, I like Travis' approach; I wish I had that on Jenkins at work.